view/SSViewer.php

```diff
@@ -512,32 +521,25 @@ public function __call($name, $arguments) {
 		if (isset($arguments[1]) && $arguments[1] != null) $params = $arguments[1]; 
 		else $params = array();
 
-		$hasInjected = $res = null;
-
-		if ($name == 'hasValue') {
-			if ($val = $this->getInjectedValue($property, $params, false)) {
-				$hasInjected = true; $res = (bool)$val['value'];
-			}
-		}
-		else { // XML_val
-			if ($val = $this->getInjectedValue($property, $params)) {
-				$hasInjected = true;
-				$obj = $val['obj'];
-				$res = $obj->forTemplate();
+		$val = $this->getInjectedValue($property, $params);
+		if ($val) {
+			$obj = $val['obj'];
+			if ($name === 'hasValue') {
+				$res = $obj instanceof Object
+					? $obj->exists()
+					: ($obj !== null);
```

Comment:
Will `$obj` ever be `null`, won't it always be an object?

It'll be a casting helper and so shouldn't we check the casting helper's `XML_val()` or similar?